lib/models/addon-discovery.js

```diff
@@ -11,7 +11,10 @@ const path = require('path');
 const resolve = require('resolve');
 const findup = require('find-up');
 const heimdall = require('heimdalljs');
-
+const PACKAGE_CACHE = Object.create(null);
```

Comment:
I'd recommend exposing a way to reset the caches, and likely invoke that when `Project.prototype.initializeAddons` is called for the first time for each project instance...